src/toolbar/Toolbar.ts

```diff
@@ -0,0 +1,159 @@
+	private _collapsed = false;
+	private _open = false;
+
+	constructor() {
+		super();
+		window.addEventListener('resize', this._collapseIfNecessary);
+	}
+
+	private _closeMenu() {
+		this._open = false;
+		this.invalidate();
+	}
+
+	private _collapseIfNecessary = () => {
+		const { collapseWidth = 800, onCollapse } = this.properties;
+		const { width } = this.meta(Dimensions).get('root').size;
+
+		if (width > collapseWidth && this._collapsed === true) {
+			this._collapsed = false;
+			onCollapse && onCollapse(this._collapsed);
+			this.invalidate();
+		}
+		else if (width <= collapseWidth && this._collapsed === false) {
+			this._collapsed = true;
+			onCollapse && onCollapse(this._collapsed);
+			this.invalidate();
+		}
+	}
+
+	private _toggleMenu() {
+		this._open = !this._open;
+		this.invalidate();
+	}
+
+	protected getFixedRootClasses(): (string | null)[] {
+		return [
+			css.rootFixed,
+			this.properties.fixed ? css.stickyFixed : null
+		];
+	}
+
+	protected getRootClasses(): (string | null)[] {
+		return [
+			css.root,
+			this._collapsed ? css.collapsed : null,
+			this.properties.fixed ? css.sticky : null
+		];
+	}
+
+	protected onAttach() {
+		this._collapseIfNecessary();
+	}
+
+	protected onDetach() {
+		window.removeEventListener('resize', this._collapseIfNecessary);
+	}
+
+	protected renderActions(): DNode {
+		const { actions = [], theme } = this.properties;
+
+		const actionsElements = actions.map((action, index) => v('div', {
+			classes: [ this.theme(css.action) ],
+			key: index
+		}, [ action ]));
+
+		if (actionsElements.length === 0) {
+			return null;
+		}
+
+		return this._collapsed ? w(SlidePane, {
+			align: Align.right,
+			closeText: 'close menu',
+			key: 'slide-pane-menu',
+			onRequestClose: this._closeMenu,
+			open: this._open,
+			theme,
+			title: 'Menu'
+		}, actionsElements) : v('div', {
+			classes: [ this.theme(css.actions), css.actionsFixed ],
+			key: 'menu'
+		}, actionsElements);
+	}
+
+	protected renderButton(): DNode {
+		return this._collapsed ? v('button', {
+			classes: [ this.theme(css.menuButton), css.menuButtonFixed ],
+			onclick: this._toggleMenu
+		}, [
+			'open menu',
+			v('i', {
+				classes: this.theme([ iconCss.icon, iconCss.barsIcon ]),
+				role: 'presentation', 'aria-hidden': 'true'
+			})
+		]) : null;
+	}
+
+	protected renderTitle(): DNode {
+		const { title } = this.properties;
+
+		return title ? v('h1', {
+			classes: [ this.theme(css.title), css.titleFixed ]
+		}, [ title ]) : null;
+	}
+
+	render(): DNode {
+		const classes = this.getRootClasses();
+		const fixedClasses = this.getFixedRootClasses();
+
+		return v('div', {
+			classes: [ ...this.theme(classes), ...fixedClasses ],
+			key: 'root'
+		}, [
+			v('div', {
+				classes: [ this.theme(css.toolbar), css.toolbarFixed ]
+			}, [
+				this.renderTitle(),
+				this.renderActions(),
+				this.renderButton()
+			]),
+			v('div', {
+				classes: [ this.theme(css.content), css.contentFixed ]
+			}, this.children)
```

Comment:
Not sure if this needs to be added now, but reversing the order of the toolbar and content based on a `position` property would be easy, and having the option of a bottom toolbar is nice. Works with the styling as-is, too.

Reply:
Added!